src/data/transform/mod.rs

```diff
@@ -99,18 +105,33 @@ pub fn serialize_entity(
     Ok(row)
 }
 
-fn get_json_value_from_string(raw_input: &str) -> serde_json::Value {
+fn get_json_value_from_string(
+    raw_input: &str,
+    column_type: &Option<ColumnType>,
+) -> anyhow::Result<serde_json::Value> {
     let raw_input_lowercase = raw_input.to_lowercase();
-    if raw_input_lowercase == "null" || raw_input.trim().is_empty() {
-        serde_json::Value::Null
-    } else if raw_input_lowercase == "true" {
-        serde_json::Value::Bool(true)
-    } else if raw_input_lowercase == "false" {
-        serde_json::Value::Bool(false)
-    } else if let Ok(number) = serde_json::Number::from_str(raw_input) {
-        serde_json::Value::Number(number)
-    } else {
-        serde_json::Value::String(raw_input.to_owned())
+
+    match (raw_input_lowercase.as_str(), column_type) {
+        (_, Some(ColumnType::String)) => Ok(serde_json::Value::String(raw_input.to_owned())),
+        (_, Some(ColumnType::Number)) =>
+            serde_json::Number::from_str(raw_input)
+                .map(serde_json::Value::Number)
+                .map_err(|_| anyhow::anyhow!("failed to convert {raw_input} into a number; make sure that you use the column types correctly")),
+        (_, Some(ColumnType::Boolean)) =>
+            raw_input.parse::<bool>()
+                .map(serde_json::Value::Bool)
+                .map_err(|_| anyhow::anyhow!("failed to convert {raw_input} into a boolean; make sure that you use the column types correctly")),
+        (input, _) if input.trim().is_empty() => Ok(serde_json::Value::Null),
+        ("null", _) => Ok(serde_json::Value::Null),
+        ("true", _) => Ok(serde_json::Value::Bool(true)),
+        ("false", _) => Ok(serde_json::Value::Bool(false)),
```

Comment:
Maybe order them like this. I think they are evaluated from top to bottom and thus more expensive checks should come later. 🤔 
```suggestion
        ("null", _) => Ok(serde_json::Value::Null),
        ("true", _) => Ok(serde_json::Value::Bool(true)),
        ("false", _) => Ok(serde_json::Value::Bool(false)),
        (input, _) if input.trim().is_empty() => Ok(serde_json::Value::Null),
```

Reply:
Seems reasonable :+1:  
I will change that.